[Merged by Bors] - Documented service discovery (#367)
stackabletech/kafka-operator: docs/modules/ROOT/pages/discovery.adoc

:clusterName: simple-kafka
:namespace: stackable
:brokerPort: 9092

= Discovery Profiles

The Stackable Operator for Apache Kafka creates a single discovery profile that allows access to the Kafka cluster https://kubernetes.io/docs/concepts/services-networking/service/[`Service`] which is published into the Kubernetes cluster as a
https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.23/#configmap-v1-core[`ConfigMap`] object.

== Profiles

Given the following Kafka cluster:

> [subs="attributes"]

Reviewer: ```suggestion
[source,yaml,subs="attributes,callouts"]
```

Author: done

----
apiVersion: kafka.stackable.tech/v1alpha1
kind: KafkaCluster
metadata:
  name: {clusterName} # <1>
  namespace: {namespace} # <2>
spec:
  [...]
----
<1> Name of the Kafka cluster which is the name of the `ConfigMap` created for this discovery profile.
<2> Namespace of the `ConfigMap` created for this discovery profile.

> The published `ConfigMap` name and namespace are equal to the cluster name and namespace. In this case `{namespace}/{clusterName}`.

Reviewer: ```suggestion
The published `ConfigMap` is created in the same namespace as the corresponding Apache Kafka cluster and has the same name. In this case `{namespace}/{clusterName}`.
```

=== Default

> This profile allows access to the Kafka cluster from inside the Kubernetes cluster, and connects directly to the metastore NodePort `Service`.

Reviewer: ```suggestion
This profile allows access to the Kafka cluster from inside the Kubernetes cluster, and connects directly to the metastore `NodePort` service.
```

== Contents

> The discovery profile contains the following fields where `{clusterName}` represents the name and `{namespace}` the namespace of the cluster:
> 
> `KAFKA`::
> ====
> Contains the thrift protocol connection string for the Kafka cluster to connect to the metastore:
> [subs="attributes"]
>   {clusterName}.{namespace}.svc.cluster.local:{brokerPort}
> ====

Reviewer: I find this a bit more pleasing and understandable:
```suggestion
The discovery profile contains the thrift protocol connection string for the Kafka cluster.

In this example:
[subs="attributes"]
----
  {clusterName}.{namespace}.svc.cluster.local:{brokerPort}
----

```
